contents/zeromq/zeromq.md

```diff
@@ -0,0 +1,97 @@
+<frontmatter>
+  title: Introduction to ZeroMQ
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+{{ booktitle | safe }}
+
+# Introduction to ZeroMQ
+
+Author: [Tan Yuanhong](https://github.com/le0tan)
+
+<box id="article-toc">
+
+* [What is ZeroMQ?](#what-is-zeromq)
+  * [What is Asynchronous Messaging and Message Queue?](#what-is-asynchronous-messaging-and-message-queue)
+* [Why ZeroMQ?](#why-zeromq)
+  * [Ease of Use](#ease-of-use)
+  * [Multi-Transport](#multi-transport)
+  * [High Performance](#high-performance)
+* [Getting Started](#getting-started)
+</box>
+
+## What is ZeroMQ?
+
+[ZeroMQ](https://zeromq.org/) is a high-performance asynchronous messaging library. It provides a message queue (MQ) for all processes communicating using the library, without a dedicated message broker - thus the *zero* in the name of *ZeroMQ*. 
+
+### What is **Asynchronous Communication** and **Message Queue**?
+
+**Asynchronous** (or async, in short) **communication** means the two (or more than two) ends of the communication don't need to be occupied at the same time to send messages. You may consider making a phone-call synchronous and sending an email async. And <tooltip content="To learn more about MQ, click [here](https://www.ibm.com/cloud/learn/message-queues)">**message queue**</tooltip> is one of the async communication protocols that puts the messages in a queue, stores them temporarily until the recipient retrieves them.
+
+Notice the word **"protocol"** used in the description above - communication protocol remains the same despite the actual medium of transport. It doesn't matter if your message is sent and received within the process between threads, or between processes, or between different physical machines across the network. It merely specifies how communicators are supposed to encode/send and decode/receive their messages.
+
+## Why ZeroMQ?
+
+**TL;DR**: By adopting ZeroMQ in your concurrent application, you're less likely to have (or at least have less) bottleneck on inter-process/thread/machine communication, while still being able to write communication logic in familiar socket APIs.
```

Comment:
Can say 'Essentially,' instead of TLDR.
Also, bottleneck can refer to many things. performance? code complexity? May need to be more specific.